cogs/TwitchAlert.py

```diff
@@ -522,58 +528,62 @@ async def loop_check_team_live(self):
-
-                            sql_update_message_id = """
-                            UPDATE UserInTwitchTeam 
-                            SET message_id = ?
-                            WHERE team_twitch_alert_id = ?
-                            AND twitch_username = ?"""
-                            self.ta_database_manager.database_manager.db_execute_commit(sql_update_message_id,
-                                                                                        args=[new_message.id, result[2],
-                                                                                              current_username])
-                    except discord.errors.Forbidden as err:
-                        logging.warning(f"TwitchAlert: {err}  Name: {channel} ID: {channel.id}")
-                        sql_remove_invalid_channel = "DELETE FROM TwitchAlerts WHERE channel_id = ?"
-                        self.ta_database_manager.database_manager.db_execute_commit(sql_remove_invalid_channel,
-                                                                                    args=[channel.id])
+            try:
+                if stream_data.get('type') == "live":
+                    current_username = str.lower(stream_data.get("user_name"))
+                    # print(current_username + " is live: "+str(usernames))
+                    usernames.remove(current_username)
+
+                    sql_find_message_id = """
+                    SELECT TITA.channel_id, UserInTwitchTeam.message_id, TITA.team_twitch_alert_id, custom_message, 
+                      default_message 
+                    FROM UserInTwitchTeam
+                    JOIN TeamInTwitchAlert TITA on UserInTwitchTeam.team_twitch_alert_id = TITA.team_twitch_alert_id
+                    JOIN TwitchAlerts TA on TITA.channel_id = TA.channel_id
+                    JOIN (SELECT extension_id, guild_id 
+                          FROM GuildExtensions 
+                          WHERE extension_id = 'TwitchAlert' OR extension_id = 'All') GE ON TA.guild_id = GE.guild_id 
+                    WHERE twitch_username = ?"""
+
+                    results = self.ta_database_manager.database_manager.db_execute_select(
+                        sql_find_message_id, args=[current_username])
+
+                    new_message_embed = None
+
+                    for result in results:
+                        channel = self.bot.get_channel(id=result[0])
+                        try:
+                            # If no Alert is posted
+                            if result[1] is None:
+                                if new_message_embed is None:
+                                    if result[3] is not None:
+                                        message = result[3]
+                                    else:
+                                        message = result[4]
+                                    new_message_embed = await self.create_alert_embed(stream_data, message)
```

Comment:
might be worth variabilizing results[0], result[1], result[3] and result[4] to make this more maintainable for the future